### ISSUE TITLE
Use with a Service Class

### ISSUE THREAD

**Issue by elado** [thoughtbot/factory_girl#720]
I have a service class

```ruby
class UserCreateService
  def self.create(params)
     user = User.new(params)
     # ...
     user.save
     # ...

     user
  end
end
```

A process of creating a user with all other needed operations is

```ruby
user = UserCreateService.create(email: "...", ...)
```

What would be a good way to integrate it with FactoryGirl?

Tried all kinds of tricks with `initialize_with` and `to_create` but whatever is being returned from `initialize_with` is the object that's being returned from `create`, and `to_create` doesn't assign the returned value as the object (as it shouldn't). The real object is initialized in the service.

So basically I need to `initialize_with` a hash but return something else.
Is there any way I can override what's a factory `create` returns?

I still want to use FactoryGirl for its RSpec integration, attribute generator, traits..

**Comment by joshuaclayton** [comment]
@elado I'm going to close this due to inactivity; I'd love to hear what you cooked up with this though!
